.github/workflows/ci.yml

```diff
@@ -1,91 +1,112 @@
 name: CI
 
 on:
-  push:
   pull_request:
-  schedule: # Trigger a job on master at midnight UTC every day
-    - cron: '0 0 * * *'
-
+  push:
+    branches:
+      - ros2
+      - master
+      - ci-dev
+ 
 jobs:
+  # amd64_build job by default runs the following in additon to building the ros2 packages:
+  #  - copyright
+  #  - cppcheck
+  #  - cpplint
+  #  - lint_cmake
+  #  - linter
+  #  - xmllint
+  #  - uncrustify
   amd64_build:
+    runs-on: ubuntu-20.04
     strategy:
-      matrix:
-        env:
-          - {ROS_DISTRO: melodic, ROS_REPO: main, CMAKE_ARGS: "-Wall -Wextra -Wpedantic -Werror", TARGET_WORKSPACE: "$TARGET_REPO_PATH metapackage_dependencies.rosinstall", UPSTREAM_WORKSPACE: "upstream_dependencies.rosinstall"}
-          - {ROS_DISTRO: melodic, ROS_REPO: testing, CMAKE_ARGS: "-Wall -Wextra -Wpedantic -Werror", TARGET_WORKSPACE: "$TARGET_REPO_PATH metapackage_dependencies.rosinstall", UPSTREAM_WORKSPACE: "upstream_dependencies.rosinstall"}
-
-    runs-on: ubuntu-18.04
+      fail-fast: false
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      - name: Run ROS Industrial CI
-        uses: ros-industrial/industrial_ci@master
-        env: ${{matrix.env}}
-
-  clang-format:
+    - name: Set Environment Variables 
+      run: |
+        echo "name=LANG" >> $GITHUB_ENV
+
+    - name: Setup ROS 2
+      uses: ros-tooling/setup-ros@0.1.4
+      with:
+        required-ros-distributions: foxy
+    
+    - name: Run Tests
+      uses: ros-tooling/action-ros-ci@0.2.0
+      with:
+        target-ros2-distro: foxy
+        uncrustify: false
+        package-name: uwrt_mars_rover
+
+  # ament_lint job will run the following linters:
+  #  - flake8 (python linter)
+  #  - clang-format (clang format check - Note: this runs clang-format-10)
+  #  Note: this job is much faster than the above job because the use of a docker container
```

Comment:
The speed difference here is actually because you arent building the code at all in this step. flake8 and clang-format dont require the code to be built to run

Reply:
From the `action-ros-lint` readme:

> This action runs ROS 2 linters on a ROS 2 package. This command does not compile any code intentionally to make it as fast as possible (<2 min). The objective is to give contributors feedback about their change quickly. It may also be used to avoid wasting CI resources (no need to compile, and run the tests if it does not pass the linters).